src/AppBundle/Controller/UserManagment/UserController.php

```diff
@@ -0,0 +1,268 @@
+            ->add("givenName", TextType::class, array(
+                "attr"=>["placeholder"=>"username"],
+                'label' => "username"))
+            ->add("clearPassword", PasswordType::class, array(
+                "attr"=>["placeholder"=>"password"],
+                'label' => "password"))
+            ->add("generatePassword", ButtonType::class, array(
+                "attr"=>[],
+                'label' => "addUser.generatePassword"))
+            ->add("generatedPassword", TextType::class, array(
+                "attr"=>["readonly"=>"",
+                    "placeholder"=>"addUser.generatedPassword"],
+                "label"=>false))
+            ->add('stamm', ChoiceType::class, array(
+                'choices'  => ArrayMethods::valueToKeyAndValue($staemme),
+            ))
+            ->add("send", SubmitType::class, array(
+                "label"=>"create",
+                "attr"=>["class"=>"btn btn-lg btn-primary btn-block"]))
+            ->getForm();
+
+        $addedSomeone = false;
+
+        //Handel the form input
+        $addUserForm->handleRequest($request);
+        if ($addUserForm->isSubmitted() && $addUserForm->isValid()) {
+            //Prepare User
+            $user->setUid($user->getGivenName());
+
+            //Create the new user
+            try {
+                $user = $userRepo->addUser($user);
+                $this->addFlash("success", "Benutzer ".$user->getUid()." hinzugefügt");
+                $addedSomeone = true;
+            } catch (UserAlreadyExistException $e) {
+                $this->addFlash("error", $e->getMessage());
+            } catch (ContextErrorException $e) {
+                $this->addFlash("error", $e->getMessage()." This probably means that this stamm (ou) does not exist.");
+            }
+        }
+
+        //Render the page
+        return $this->render("userManagment/addUser.html.twig", array(
+            "addAUserForm" => $addUserForm->createView(),
+            "addedPerson" => $user,
+            "addedSomeone" => $addedSomeone
+        ));
+    }
+
+    /**
+     * @Route("/users/detail", name="detailUser")
+     * @param Request $request
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function showDetailUser(Request $request)
+    {
+        $loggedInUser = $this->get('security.token_storage')->getToken()->getUser();
+        $userRepo = $this->get("data.userRepository");
+
+        if ($request->get("uid", $loggedInUser->getUid()) != $loggedInUser->getUid()) {
+            $this->denyAccessUnlessGranted("ROLE_elder", null, "You are not allowed to see this user");
+
+            $userToShow = $userRepo->getUserByUid($request->get("uid"));
+        } else {
+            $userToShow = $loggedInUser;
+        }
+
+        $editUserForm = false;
+        //is the user allowed to edit this user?
+        //TODO add the ability for groupleaders to edit their users
+        if ($userToShow->getUid() == $loggedInUser->getUid()
```

Comment:
Dieses lange if Statement ist nicht einfach überschaubar. Abfrage in eine eigene Methode auslagern!?